azurerm/resource_arm_eventgrid_event_subscription.go

```diff
@@ -0,0 +1,645 @@
+				Required: true,
+				ForceNew: true,
+			},
+
+			"scope": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"event_delivery_schema": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(eventgrid.EventGridSchema),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(eventgrid.CloudEventV01Schema),
+					string(eventgrid.CustomInputSchema),
+					string(eventgrid.EventGridSchema),
+				}, false),
+			},
+
+			"topic_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"storage_queue_endpoint": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: []string{"eventhub_endpoint", "hybrid_connection_endpoint", "webhook_endpoint"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"storage_account_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"queue_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"eventhub_endpoint": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: []string{"storage_queue_endpoint", "hybrid_connection_endpoint", "webhook_endpoint"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"eventhub_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"hybrid_connection_endpoint": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: []string{"storage_queue_endpoint", "eventhub_endpoint", "webhook_endpoint"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"hybrid_connection_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"webhook_endpoint": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: []string{"storage_queue_endpoint", "eventhub_endpoint", "hybrid_connection_endpoint"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"url": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"included_event_types": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"subject_filter": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"subject_begins_with": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"subject_ends_with": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"case_sensitive": {
+							Type:     schema.TypeBool,
+							Optional: true,
```

Comment:
by being Optional this implicitly has a default of `false`, which is may be worth highlighting in the docs

Reply:
Done